### DISCUSSION TITLE
How do I install and use this on Gitpod?

### DISCUSSION THREAD

**Discussion by necroheart31** [titaniumnetwork-dev/Corrosion#16] [Q&A]
I have been using Alloy for over half a year now, and recently I saw that it was removed and changed into Corrosion. Alloy was intuitive and directly told you how to use it, but this makes no sense. I have it as a workspace on Gitpod and installed it, but I don't know how to start it. I also want to say that I'm on a Chromebook.

**Comment by QuiteAFancyEmerald** [comment]
Like the updated version of Alloy, Corrosion is a module. You can't simply "deploy" or make a web app from this repo without proper setup. Simply make a workspace by cloning this repository with Gitpod as usual and navigate to the demo folder then do the usual setup. I would look into also following the pm2 steps explained in the README.

**Useful commands after cloning:**
```
cd demo
npm install
node index.js
```
Corrosion should be running now! But this isn't really enough for a production setup. Now you would want to look into setting up nginx and pm2. I'll provide the basics for pm2 below.

**PM2 setup:**
```
npm i pm2 -g
pm2 start index.js
pm2 startup
pm2 save
```

**Reply by necroheart31** [reply]
I used cd demo and npm install, but when I used node index.js, I recieved this error:
node:events:368
      throw er; // Unhandled 'error' event
      ^

Error: listen EACCES: permission denied 0.0.0.0:443
    at Server.setupListenHandle [as _listen2] (node:net:1317:21)
    at listenInCluster (node:net:1382:12)
    at Server.listen (node:net:1469:7)
    at Object.<anonymous> (/workspace/Corrosion/demo/index.js:19:119)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1361:8)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  code: 'EACCES',
  errno: -13,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 443
}

**Comment by QuiteAFancyEmerald** [comment]
Also change the port for it! :D

**Reply by necroheart31** [reply]
After changing the port, running "node index.js" returned "bundled scripts" twice, and nothing else happened. Am I doing something wrong?

**Comment by nevr13** [comment]
Make sure that you correct the ssl.key, ssl.cert, and possibly the const Corrosion = require('../'); locations.

```
const https = require('https');
const fs = require('fs');
const path = require('path');
const ssl = {
    **key: fs.readFileSync(path.join(__dirname, '/ssl.key')),
    cert: fs.readFileSync(path.join(__dirname, '/ssl.cert')),**
};
const server = https.createServer(ssl);
**const Corrosion = require('../');**
const proxy = new Corrosion({
    codec: 'xor',
});

**proxy.bundleScripts();**

server.on('request', (request, response) => {
    if (request.url.startsWith(proxy.prefix)) return proxy.request(request, response);
    response.end(fs.readFileSync(__dirname + '/index.html', 'utf-8'));
}).on('upgrade', (clientRequest, clientSocket, clientHead) => proxy.upgrade(clientRequest, clientSocket, clientHead)).listen(443);
```
